app/controllers/planning_applications_controller.rb

```diff
@@ -71,14 +71,37 @@ def withdraw
     redirect_to @planning_application
   end
 
+  def cancel_confirmation
+    render :cancel_confirmation
+  end
+
+  def cancel
+    status = authorize_user_can_update_status(
+        planning_application_params[:status])
+    apply_cancellation(status)
+    @planning_application.update!(cancellation_comment: planning_application_params[:cancellation_comment])
```

Comment:
Could we add the cancellation_comment as an optional parameter to the `withdrawn` and `returned` aasm events? Something like this? https://github.com/aasm/aasm/issues/506#issuecomment-532434609